plugin-modernizer-core/src/test/java/io/jenkins/tools/pluginmodernizer/core/extractor/MetadataCollectorTest.java

```diff
@@ -164,4 +164,226 @@ void testPlugin() throws Exception {
+                              <name>jenkinsci-snapshots</name>
+                              <url>https://repo.jenkins-ci.org/snapshots</url>
+                            </snapshotRepository>
+                          </distributionManagement>
+
+                          <properties>
+                            <revision>1.8.1</revision>
+                            <java.level>8</java.level>
+                            <changelist>-SNAPSHOT</changelist>
+                            <jenkins.version>2.426.3</jenkins.version>
+                            <spotbugs.effort>Max</spotbugs.effort>
+                            <spotbugs.threshold>Low</spotbugs.threshold>
+                            <gitHubRepo>jenkinsci/${project.artifactId}</gitHubRepo>
+                            <hpi.compatibleSinceVersion>1.4.0</hpi.compatibleSinceVersion>
+                            <mockserver.version>5.15.0</mockserver.version>
+                            <spotless.check.skip>false</spotless.check.skip>
+                          </properties>
+
+                          <dependencyManagement>
+                            <dependencies>
+                              <dependency>
+                                <!-- Pick up common dependencies for the selected LTS line: https://github.com/jenkinsci/bom#usage -->
+                                <groupId>io.jenkins.tools.bom</groupId>
+                                <artifactId>bom-2.414.x</artifactId>
+                                <version>2950.va_633b_f42f759</version>
+                                <type>pom</type>
+                                <scope>import</scope>
+                              </dependency>
+                            </dependencies>
+                          </dependencyManagement>
+
+                          <dependencies>
+                            <dependency>
+                              <groupId>io.jenkins.plugins</groupId>
+                              <artifactId>caffeine-api</artifactId>
+                            </dependency>
+                          </dependencies>
+
+                          <repositories>
+                            <repository>
+                              <id>repo.jenkins-ci.org</id>
+                              <url>https://repo.jenkins-ci.org/public/</url>
+                            </repository>
+                          </repositories>
+
+                          <pluginRepositories>
+                            <pluginRepository>
+                              <id>repo.jenkins-ci.org</id>
+                              <url>https://repo.jenkins-ci.org/public/</url>
+                            </pluginRepository>
+                          </pluginRepositories>
+                        </project>
+                        """));
+        PluginMetadata pluginMetadata = new PluginMetadata().refresh();
+        JDK jdkVersion = pluginMetadata.getJdkVersion();
+        assertEquals(JDK.JAVA_17, jdkVersion);
+    }
```

Comment:
Missing some test with empty buildPlugin() or more outdated/exotic configuration.

For example how the visitors are reacting to https://github.com/jenkinsci/forensics-api-plugin/blob/main/Jenkinsfile